### PR TITLE
added sfpr_structure_size_type

### PR DESCRIPTION
**Pull request by lucy-schick**
added `sfpr_structure_size_type` from fraser as per https://github.com/NewGraphEnvironment/fish_passage_template_reporting/issues/32.

PR Ready to go.

### CODE REVIEW

File: inst/staticexports/ng_fpr.R [hunk 1]
```diff
@@ -165,3 +165,89 @@ sfpr_create_hydrograph <- function(
 }
 
 
+#' Determine replacement structure type and size based on measured field metrics.
+#' @param dat PSCIS data
+#' @param fill_dpth standard fill depth, default is 3m.
+#' @param brdg_wdth standard bridge width, default is 15m.
+#' @param chn_wdth_max maximum channel width where the bridge should start to be more than brdg_wdth, default is brdg_wdth - 5m.
```
Comment (NewGraphEnvironment) [inline inst/staticexports/ng_fpr.R:172]: ill try to clarify why this is what it is in subsequent commits

File: inst/staticexports/ng_fpr.R [hunk 2]
```diff
@@ -165,3 +165,89 @@ sfpr_create_hydrograph <- function(
...
+#'
+#' #' @examples \dontrun{
+#' fpr_structure_size_type(dat)
+#' }
+#'
+sfpr_structure_size_type <- function(
+    dat = NULL,
+    fill_dpth = 3,
+    brdg_wdth = 15,
+    chn_wdth_max = brdg_wdth - 5,
```
Comment (NewGraphEnvironment) [inline inst/staticexports/ng_fpr.R:190]: including a param in a param is not something i have seen done before i odn't think. ill prob change and put that calc in the code of the funciton

File: inst/staticexports/ng_fpr.R [hunk 3]
```diff
@@ -165,3 +165,89 @@ sfpr_create_hydrograph <- function(
...
+    fill_dpth_mult = 3) {
+
+  if (is.null(dat))
+    stop('please provide "dat" (dataframe) object')
+  if (!is.data.frame(dat))
```
Comment (NewGraphEnvironment) [inline inst/staticexports/ng_fpr.R:195]: can use chk for htis

**Review by NewGraphEnvironment — COMMENTED** [review]
was hopeing to let the template excel back and forths go but that is prob  more trouble than it is worth in 24/25. I will do a few changes to simlify the code after merging. maybe have a look at the changes in the commit after the pr

File: inst/staticexports/ng_fpr.R (hunk 2 above)
Comment (lucy-schick) [reply inst/staticexports/ng_fpr.R:190]: okay sure. Probably safer that way.

**Pull request merged**